### DISCUSSION TITLE
related().attach() doesn't check existance of relation (many to many)

### DISCUSSION THREAD

**Discussion by guoyunhe** [adonisjs/core#4065] [Help] [answered]
I have this code to add new relationship:

```js
post.related('images').attach([imageId]);
```

I thought that it would check existance before creating new relationship. But it throws error when inserting duplicate images:

```
  insert into `post_image` (`post_id`, `image_id`) values (14, '27') - Duplicate entry '14-27' for key 'post_image.post_image_post_id_image_id_unique'
```

Do I have to query existing relationships before adding new ones?

**Comment by RomainLanz** [comment, marked as answer]
Hey @guoyunhe! 👋🏻 

Use `sync` instead.

```ts
post.related('images').sync([imageId], false);
```

📚 https://docs.adonisjs.com/reference/orm/relations/many-to-many#sync
